src/fudge/log.clj

```diff
@@ -20,6 +20,15 @@
   [c k & args]
   (apply (k c) c args))
 
+(defmacro calling-meta
+  "Used by other macros to get the namespace, file and line of the log call.
+   Builds a def with a generated symbol to expand and evaluate for the info."
```

Comment:
`defonce` would probably be better for loops, yes.

As for the defs everywhere where logging, I did spend quite a while in query console and writing extra output from the test file to check the current vars in `(ns-publics (all-ns))` and anywhere else I could think to check. I couldn't find these defs cluttering up the namespace anywhere. The use of `(gen-sym)` also means that the defs are unique and won't clash with user defs, so even if they *do* stay around somewhere I don't think have one additional randomly named def with no value per log call is an issue.